spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkConfParser.java

```diff
@@ -106,7 +106,7 @@ public int parse() {
     }
 
     public Integer parseOptional() {
-      return parse(Integer::parseInt, null);
+      return parse(Integer::parseInt, defaultValue);
```

Comment:
If a default value was provided explicitly, it should be used.

Reply:
Why not throw an exception? Seems like these cases are mutually exclusive.

Reply:
Oh, I see. You can have a `null` default for objects like String. I think that's fine for String but when the `defaultValue` is set as a primitive `long` or `int` I think we don't need to support `parseOptional` and require `parse`.

Reply:
Yeah, we can probably do this for String only. There are table properties with explicit NULL default values.

Reply:
Well, `parseOptional` does produce `Integer` so if we have a default value that is null, it may still apply.

```
public static final String PARQUET_COMPRESSION_LEVEL_DEFAULT = null;
```

We can hit such cases for ints as well, I assume.